network/multiplexer.go

```diff
@@ -57,6 +57,15 @@ func (m *Multiplexer) getHandler(tag Tag) (MessageHandler, bool) {
 	return nil, false
 }
 
+// wantsTag returns true if there is a registered handler for the specified tag
+func (m *Multiplexer) wantsTag(tag Tag) bool {
+	if handlers := m.getHandlersMap(); handlers != nil {
```

Comment:
this costs one atomic.Value.Load() and map typecast per message, I guess we are already doing 1 per message so this would make another. You only need this check when a test is testing a non-standard protocol — maybe in the tests that need to use custom handlers we add an init() in e.g. wsNetwork_test.go that enables tag-checking ..?

Reply:
like https://github.com/algorand/go-algorand/pull/4517/commits/06f5c560298ae823105169dbe0a79cd8f9681af9